lib/loader.js

```diff
@@ -107,22 +109,21 @@ function createBot (options = {}) {
   if (!bot._client.wait_connect) next()
   else bot._client.once('connect_allowed', next)
   function next () {
-    bot.registry = require('prismarine-registry')(bot._client.version)
-    const version = bot.registry.version
-    if (supportedVersions.indexOf(version.majorVersion) === -1) {
-      throw new Error(`Version ${version.minecraftVersion} is not supported.`)
-    }
+    const serverPingVersion = bot._client.version
+    bot.registry = require('prismarine-registry')(serverPingVersion)
+    if (!bot.registry?.version) throw new Error(`Server version '${serverPingVersion}' is not supported, no data for version`)
 
-    const latestTestedVersion = testedVersions[testedVersions.length - 1]
-    const latestProtocolVersion = require('prismarine-registry')(latestTestedVersion).protocolVersion
-    if (version.protocolVersion > latestProtocolVersion) {
-      throw new Error(`Version ${version.minecraftVersion} is not supported. Latest supported version is ${latestTestedVersion}.`)
+    const versionData = bot.registry.version
+    if (versionData.version > latestSupportedVersionData.version) {
```

Comment:
how is that operator working?

Reply:
versionData is a version object in minecraft-data, https://github.com/PrismarineJS/node-minecraft-data/blob/master/doc/api.md#mcdataversionversion. The `.version` is the protocol version.

Reply:
is that using https://github.com/PrismarineJS/node-minecraft-data/blob/master/index.js#L44 ?

Reply:
No, those functions check against dataVersion, this check was supposed to check protocol version but the previous logic was wrong

Reply:
I think protocol version is not enough, we should check against the version ordering in minecraft data

sometimes mojang does not increase the protocol version but makes mineflayer-level breaking changes

Reply:
why not use the operators defined in nmd ?

Reply:
ok, updated to use dataVersion comparison functions in node-mcdata